docs/query-builder.md

```diff
@@ -448,64 +448,161 @@ DB::collection('items')
 $user->push('items', 'boots');
+$user->push('items', ['hat', 'jeans']);
 ```
 
+To add an array as a value to the messages array. value arrays must be nested into a list array, otherwise they are
+interpreted as a list of scalar values.
+
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // Push an array as a value to the items array
+  ->push('messages', [
+      [
+          'from'    => 'Jane Doe',
+          'message' => 'Hi John',
+      ],
+  ]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
+
+// Or
 
 $user->push('messages', [
-    'from' => 'Jane Doe',
-    'message' => 'Hi John',
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
 ]);
 ```
 
-If you **DON'T** want duplicate items, set the third parameter to `true`:
+If you **DON'T** want duplicate values, set the third parameter to `true`:
 
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots', true);
+  ->where('name', 'John')
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  ->push('items', ['hat', 'boots', 'jeans'], true);
+// Result:
+// items: ['boots', 'hat', 'jeans']
+
+// Or
+
+$user->push('messages', [
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
+]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
 
-$user->push('items', 'boots', true);
+$user->push('messages', [
+    [
+        'from'    => 'Jess Doe',
+        'message' => 'Hi',
+    ],
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
+], true);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//      { from: "Jess Doe", message: "Hi" }
+//  ]
 ```
 
 **Pull**
 
-Remove an item from an array.
+Remove one or multiple values form items array.
 
 ```php
+// items: ['boots', 'hat', 'jeans']
+
 DB::collection('users')
-    ->where('name', 'John')
-    ->pull('items', 'boots');
+  ->where('name', 'John')
+  ->pull('items', 'boots'); // Pull a single value
+// Result:
+// items: ['hat', 'jeans']
+
+// Or pull multiple values
 
-$user->pull('items', 'boots');
+$user->pull('items', ['boots', 'jeans']);
+// Result:
+// items: ['hat']
 ```
 
+Also, you can pull an array value from the messages array.
+
 ```php
+// Latest state:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//      { from: "Jess Doe", message: "Hi" }
+//  ]
+
 DB::collection('users')
-    ->where('name', 'John')
-    ->pull('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // It will search for the exact array and pulls it
+  ->pull('messages', [
+      [
+          'from' => 'Jane Doe',
+          'message' => 'Hi John',
+      ]
```

Comment:
Consider writing all of these embedded documents as one-liners through this PR to make the examples more concise. This would be particularly helpful for the example where a third boolean argument is provided to `push()` since it can easily get lost at the end of all of the whitespace in the method call.